rust/datafusion/src/sql/planner.rs

```diff
@@ -524,25 +524,18 @@ impl<'a, S: SchemaProvider> SqlToRel<'a, S> {
                             args: rex_args,
                         })
                     }
-                    // finally, built-in scalar functions
+                    // finally, user-defined functions
                     _ => match self.schema_provider.get_function_meta(&name) {
                         Some(fm) => {
-                            let rex_args = function
+                            let args = function
                                 .args
                                 .iter()
                                 .map(|a| self.sql_to_rex(a, schema))
                                 .collect::<Result<Vec<Expr>>>()?;
 
-                            let mut safe_args: Vec<Expr> = vec![];
```

Comment:
the sql planner no longer needs to add casts based on the function's signature. This was causing statements such as `SELECT my_sqrt(c1) FROM t` to be logically planned as `SELECT my_sqrt(CAST(c1 AS f64)) FROM t`, which is counter-intuitive when reading the result.